### PR TITLE
Several improvements and bug fixes

### PR DESCRIPTION
**Pull request by OscarDPan**
@danielenricocahall 
I made several minor improvements on my fork, could you review them when you have time?

I am pretty new to the open-source code contribution thing, I am open to all suggestions :)

### CODE REVIEW

**Comment by danielenricocahall** [comment]
Hello @OscarDPan ! Thank you for your contribution! I will review things very soon!

File: elephas/spark_model.py [hunk 1]
```diff
@@ -181,8 +180,10 @@ def _fit(self, rdd: RDD, **kwargs):
                                  optimizer, loss, metrics, custom)
             gradients = rdd.mapPartitions(worker.train).collect()
             new_parameters = self._master_network.get_weights()
-            for grad in gradients:  # simply accumulate gradients one by one
-                new_parameters = subtract_params(new_parameters, grad)
+            number_of_sub_models = len(gradients)
+            for grad in gradients:  # Accumulate simple average gradients one by one
+                weighted_grad = divide_by(grad, number_of_sub_models)
+                new_parameters = subtract_params(new_parameters, weighted_grad)
```
Comment (danielenricocahall) [inline elephas/spark_model.py:186]: Parsing through some of the literature (https://arxiv.org/pdf/1802.09941.pdf, https://arxiv.org/pdf/2007.03970.pdf, http://cs229.stanford.edu/proj2012/Percy-ParallelLearningContentRecommendationsUsingMapReduce.pdf), I think this is correct. This isn't a drastic change, either, as we're not changing the direction of the gradients, just the magnitude (scaling them down by `number_of_submodels`). I would be fine with merging this in, but before I do, I just want to run it by @maxpumperla , as he originally implemented this, and he may have differing views on this.

**Review by danielenricocahall — COMMENTED** [review]
Overall looks good to me! Just want @maxpumperla 's feedback/approval regarding the update to the synchronous algorithm.

File: elephas/spark_model.py (hunk 1 above)
Comment (maxpumperla) [reply elephas/spark_model.py:186]: makes a lot of sense, in the end this is the reason I wrote the `divide_by` util. I can't account for why this was missing, but this change is certainly correct.

**Pull request merged**